src/components/Navigation/BottomTabBar/index.tsx

```diff
@@ -79,6 +79,33 @@ function BottomTabBar({selectedTab, isTooltipAllowed = false}: BottomTabBarProps
         clearSelectedText();
         interceptAnonymousUser(() => {
             const defaultCannedQuery = buildCannedSearchQuery();
+
+            const rootState = navigationRef.getRootState() as State<RootNavigatorParamList>;
+            const lastSearchNavigator = rootState.routes.findLast((route) => route.name === NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR);
```

Comment:
Here's the fix for preserving state when navigating from BottomBar:
```suggestion
const lastSearchNavigator = rootState.routes.findLast((route) => route.name === NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR);
            const lastSearchNavigatorState = lastSearchNavigator && lastSearchNavigator.state && lastSearchNavigator.key ? getPreservedSplitNavigatorState(lastSearchNavigator?.key) : undefined;
            const lastSearchRoute = lastSearchNavigatorState?.routes.findLast((route) => route.name === SCREENS.SEARCH.ROOT); 
```

This solution came from @WojtekBoman - thanks!

A few words about it:
previously Search was "flat" so we didn't need to do any extra steps to try and preserve it's state. The route with `params` was on the top level of navigation so it worked.
Now we need this extra steps of finding the last search route within the navigator to grab the query.

If you are able to improve the dirty logic expression that I did `lastSearchNavigator && lastSearchNavigator.state && lastSearchNavigator.key` then feel free 😅